--- configure.ac	2014-07-15 15:49:54.801853422 -0400
+++ configure.ac	2014-07-15 15:50:18.625057694 -0400
@@ -40,8 +40,8 @@
 # Checks for library functions.
 AC_FUNC_VPRINTF
 AC_FUNC_MEMCMP
-AC_FUNC_MALLOC
-AC_FUNC_REALLOC
+# AC_FUNC_MALLOC
+# AC_FUNC_REALLOC
 AC_CHECK_FUNCS(strcasecmp strdup strerror snprintf vsnprintf vasprintf open vsyslog strncasecmp setlocale)
 AC_CHECK_DECLS([INFINITY], [], [], [[#include <math.h>]])
 AC_CHECK_DECLS([nan], [], [], [[#include <math.h>]])
--- random_seed.c	2014-07-15 15:50:11.958427143 -0400
+++ random_seed.c	2014-07-15 15:50:29.554998481 -0400
@@ -181,7 +181,7 @@
 #define HAVE_CRYPTGENRANDOM 1
 
 #include <windows.h>
-#pragma comment(lib, "advapi32.lib")
+/* #pragma comment(lib, "advapi32.lib") */
 
 static int get_cryptgenrandom_seed()
 {
