--- lgi/Makefile.orig	2015-08-02 19:50:39.824191895 -0400
+++ lgi/Makefile	2015-08-02 19:51:52.498649119 -0400
@@ -5,32 +5,20 @@
 # License: MIT
 #
 
-PREFIX = /usr/local
+PREFIX ?= /usr/local
 HOST_OS = $(shell uname -s | tr A-Z a-z)
-LUA_VERSION=5.1
-LUA_LIBDIR = $(PREFIX)/lib/lua/$(LUA_VERSION)
-LUA_SHAREDIR = $(PREFIX)/share/lua/$(LUA_VERSION)
+LUA_VERSION ?= 5.1
+LUA_LIBDIR ?= $(PREFIX)/lib/lua/$(LUA_VERSION)
+LUA_SHAREDIR ?= $(PREFIX)/share/lua/$(LUA_VERSION)
 
 PKG_CONFIG = pkg-config
 GINAME = gobject-introspection-1.0
 PKGS = $(GINAME) gmodule-2.0 libffi
 VERSION_FILE = version.lua
 
-ifneq ($(filter cygwin% msys% mingw%, $(HOST_OS)),)
 CORE = corelgilua51.dll
 LIBFLAG = -shared
-LIBS += -llua
-else
-ifeq ($(HOST_OS),darwin)
-CORE = corelgilua51.so
-LIBFLAG = -bundle -undefined dynamic_lookup
-CCSHARED = -fno-common
-else
-CORE = corelgilua51.so
-LIBFLAG = -shared
-CCSHARED = -fPIC
-endif
-endif
+LIBS = $(LUA_DLL)
 
 OBJS = buffer.o callable.o core.o gi.o marshal.o object.o record.o
 
@@ -81,6 +69,7 @@
 	mkdir -p $(DESTDIR)$(LUA_SHAREDIR)
 	cp ../lgi.lua $(DESTDIR)$(LUA_SHAREDIR)
 	mkdir -p $(DESTDIR)$(LUA_SHAREDIR)/lgi
-	cp $(CORESOURCES) $(VERSION_FILE) $(DESTDIR)$(LUA_SHAREDIR)/lgi
+	cp $(CORESOURCES) $(DESTDIR)$(LUA_SHAREDIR)/lgi
 	mkdir -p $(DESTDIR)$(LUA_SHAREDIR)/lgi/override
 	cp $(OVERRIDES) $(DESTDIR)$(LUA_SHAREDIR)/lgi/override
+
