--- lgi/Makefile.orig	2015-01-06 14:38:50.530671206 +0000
+++ lgi/Makefile	2015-01-06 14:39:33.072595481 +0000
@@ -5,46 +5,30 @@
 # License: MIT
 #
 
-PREFIX = /usr/local
-LUA_VERSION=5.1
-LUA_LIBDIR = $(PREFIX)/lib/lua/$(LUA_VERSION)
-LUA_SHAREDIR = $(PREFIX)/share/lua/$(LUA_VERSION)
+PREFIX ?= /usr/local
+LUA_VERSION ?= 5.1
+PKG_CONFIG ?= pkg-config
+LUA_LIBDIR ?= $(PREFIX)/lib/lua/$(LUA_VERSION)
+LUA_SHAREDIR ?= $(PREFIX)/share/lua/$(LUA_VERSION)
+
+GIR = gobject-introspection-1.0
+GMODULE = gmodule-2.0
+LIBFFI = libffi
+PKGS := $(GIR) $(GMODULE) $(LIBFFI)
+CFLAGS := -Wall -Wextra -O2 -g -DGLIB_STATIC_COMPILATION=1 $(shell $(PKG_CONFIG) --cflags $(PKGS)) $(CFLAGS)
+LIBS := $(shell $(PKG_CONFIG) --static $(PKGS)) $(LIBS)
+LDFLAGS := -static $(LDFLAGS)
 
-GINAME = gobject-introspection-1.0
-PKGS = $(GINAME) gmodule-2.0 libffi
 VERSION_FILE = version.lua
-
-ifneq ($(filter CYGWIN%, $(shell uname -s)),)
-CORE = corelgilua51.dll
-LIBFLAG = -shared
-LIBS += -llua
-else
-ifeq ($(shell uname -s),Darwin)
-CORE = corelgilua51.so
-LIBFLAG = -bundle -undefined dynamic_lookup
-CCSHARED = -fno-common
-else
-CORE = corelgilua51.so
-LIBFLAG = -shared
-CCSHARED = -fPIC
-endif
-endif
-
+CORE = liblgi.a
 OBJS = buffer.o callable.o core.o gi.o marshal.o object.o record.o
-
-ifndef CFLAGS
-ifndef COPTFLAGS
-CFLAGS = -Wall -Wextra -O2 -g
-endif
-endif
-ALL_CFLAGS = $(CCSHARED) $(COPTFLAGS) $(LUA_CFLAGS) $(shell pkg-config --cflags $(PKGS)) $(CFLAGS)
-LIBS += $(shell pkg-config --libs $(PKGS))
-ALL_LDFLAGS = $(LIBFLAG) $(LDFLAGS)
 DEPCHECK = .depcheck
 
 # Precondition check
 $(DEPCHECK) : Makefile
-	pkg-config --exists '$(GINAME) >= 0.10.8' --print-errors
+	pkg-config --exists '$(GIR) >= 0.10.8' --print-errors
+	pkg-config --exists '$(GMODULE) >= 2.42.1' --print-errors
+	pkg-config --exists '$(LIBFFI) >= 3.2.1' --print-errors
 	touch $@
 
 .PHONY : all clean install
@@ -54,10 +38,11 @@
 	rm -f $(CORE) $(OBJS)
 
 %.o : %.c
-	$(CC) $(ALL_CFLAGS) -c -o $@ $<
+	$(CC) $(CFLAGS) -c -o $@ $<
 
 $(CORE) : $(OBJS)
-	$(CC) $(ALL_LDFLAGS) -o $@ $(OBJS) $(LIBS)
+	$(AR) crv $(CORE) $(OBJS)
+	$(RANLIB) $(CORE)
 
 $(VERSION_FILE) : Makefile ../Makefile
 	echo "return '$(VERSION)'" > $@
@@ -75,10 +60,10 @@
 
 install : $(CORE) $(VERSION_FILE)
 	mkdir -p $(DESTDIR)$(LUA_LIBDIR)/lgi
-	cp $(CORE) $(DESTDIR)$(LUA_LIBDIR)/lgi
+	cp $(CORE) $(PREFIX)/lib
 	mkdir -p $(DESTDIR)$(LUA_SHAREDIR)
 	cp ../lgi.lua $(DESTDIR)$(LUA_SHAREDIR)
 	mkdir -p $(DESTDIR)$(LUA_SHAREDIR)/lgi
-	cp $(CORESOURCES) $(VERSION_FILE) $(DESTDIR)$(LUA_SHAREDIR)/lgi
+	cp $(CORESOURCES) $(DESTDIR)$(LUA_SHAREDIR)/lgi
 	mkdir -p $(DESTDIR)$(LUA_SHAREDIR)/lgi/override
 	cp $(OVERRIDES) $(DESTDIR)$(LUA_SHAREDIR)/lgi/override
